feature/INTHAD-216-Map-export(EXPERIMENTAL)

shanmukhananda opened the pull request.

shanmukhananda committed:
export Map, use CHOLMOD solver

paul-michalik reviewed (commented):
I suggest to use the ".vendor" strategy for copying/downloading (but not tracking) stuff from 3rd party libraries. This is now being integrated on the data source branch... Hm. let's keep this as it is for now.

paul-michalik commented on experiments/cpp/slam-runner/main.cpp:
```diff
@@ -0,0 +1,124 @@
+    const char nl{'\n'};
+    const int precision{15};
+
+    auto kfmap = std::get<ext::keyframe_map_t>(map);
```
It is better to use `std::tie` before we get structured bindings:
```
std::tie(a, b, c) = map; 
```

paul-michalik commented:
Hm. Looks like this is a mix of map export and covariance recovery, or am I mistaken?

shanmukhananda commented:
Yes that's right, I have mixed map-export changes, cholmod solver cov recovery changes to this branch. I found it easier to keep all the changes in one place, as I was facing strange build issues across different platforms.

I would like to use this branch/PR to discuss the changes required to recover the co-variance, after this I will make separate discrete PRs.

paul-michalik commented:
Ok, go ahead!

Sent from my Windows 10 phone

________________________________
From: Shanmukha <notifications@github.com>
Sent: Thursday, August 16, 2018 12:05:00 PM
To: paul-michalik/ORB_SLAM2
Cc: Paul Michalik; Comment
Subject: Re: [paul-michalik/ORB_SLAM2] feature/INTHAD-216-Map-export (#93)


Yes that's right, I have mixed map-export changes, cholmod solver cov recovery changes to this branch. I found it easier to keep all the changes in one place, as I was facing strange build issues across different platforms.

I would like to use this branch/PR to discuss the changes required to recover the co-variance, after this I will make separate discrete PRs.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpaul-michalik%2FORB_SLAM2%2Fpull%2F93%23issuecomment-413492936&data=02%7C01%7C%7C5469c46024a34a38f5a608d6035fbaaa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636700107016375339&sdata=JzEWwADkpoc4GwST1M%2Fw3CUO3rF9fGNOz58LYPFu%2Be8%3D&reserved=0>, or mute the thread<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAKF8blEbEIWngE-6kTIktTlXl7FP3im1ks5uRUPMgaJpZM4VoJHI&data=02%7C01%7C%7C5469c46024a34a38f5a608d6035fbaaa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636700107016375339&sdata=XPA5ZXyUKFTNkUZO9TvJk5QSA25nIn19eUlajoLvgs4%3D&reserved=0>.

shanmukhananda committed:
disable cov calculation

added mono grayscale + depth image runner

temporary commit

enable all options in build_.bat

use BlockSolverX
